.github/workflows/tests.yml

```diff
@@ -0,0 +1,53 @@
+name: Reportengine tests
+
+on: [push]
+
+jobs:
+  test:
+    runs-on: ubuntu-latest
+
+    strategy:
+      fail-fast: false
+      matrix:
+        python-version: ["3.12"]
+
+    steps:
+    - name: Checkout repository
+      uses: actions/checkout@v2
+
+    - name: Set up Python ${{ matrix.python-version }}
+      uses: actions/setup-python@v2
+      with:
+        python-version: ${{ matrix.python-version }}
+    
+    - name: Setup Miniconda
+      uses: conda-incubator/setup-miniconda@v3
+      with:
+        activate-environment: test
+        use-mamba: true
+
+    - name: Initialize Conda
+      shell: bash -l {0}
+      run: |
+        conda init bash
+        source ~/.bashrc
```

Comment:
Looking at the github page, this shouldn't be necessary: https://github.com/conda-incubator/setup-miniconda